apps/dash-uber-rides-demo/app.py

```diff
@@ -0,0 +1,491 @@
+        if i in xSelected and len(xSelected) < 24:
+            colorVal[i] = "#FFFFFF"
+        xVal.append(i)
+        # Get the number of rides at a particular time
+        yVal.append(len(totalList[month][day][totalList[month][day].index.hour == i]))
+    return [np.array(xVal), np.array(yVal), np.array(colorVal)]
+
+
+# Selected Data in the Histogram updates the Values in the DatePicker
+@app.callback(
+    Output("bar-selector", "value"),
+    [Input("histogram", "selectedData"), Input("histogram", "clickData")],
+)
+def update_bar_selector(value, clickData):
+    holder = []
+    if clickData:
+        holder.append(str(int(clickData["points"][0]["x"])))
+    if value:
+        for x in value["points"]:
+            holder.append(str(int(x["x"])))
+    return list(set(holder))
+
+
+# Clear Selected Data if Click Data is used
+@app.callback(Output("histogram", "selectedData"), [Input("histogram", "clickData")])
+def update_selected_data(clickData):
+    if clickData:
+        return {"points": []}
+
+
+# Update the total number of rides Tag
+@app.callback(Output("total-rides", "children"), [Input("date-picker", "date")])
+def update_total_rides(datePicked):
+    date_picked = dt.strptime(datePicked, "%Y-%m-%d")
+    return "Total Number of rides: {:,d}".format(
+        len(totalList[date_picked.month - 4][date_picked.day - 1])
+    )
+
+
+# Update the total number of rides in selected times
+@app.callback(
+    [Output("total-rides-selection", "children"), Output("date-value", "children")],
+    [Input("date-picker", "date"), Input("bar-selector", "value")],
+)
+def update_total_rides_selection(datePicked, selection):
+    firstOutput = ""
+
+    if selection is not None or len(selection) is not 0:
+        date_picked = dt.strptime(datePicked, "%Y-%m-%d")
+        totalInSelection = 0
+        for x in selection:
+            totalInSelection += len(
+                totalList[date_picked.month - 4][date_picked.day - 1][
+                    totalList[date_picked.month - 4][date_picked.day - 1].index.hour
+                    == int(x)
+                ]
+            )
+        firstOutput = "Total rides in selection: {:,d}".format(totalInSelection)
+
+    holder = []
+    if (
+        datePicked is None
+        or selection is None
+        or len(selection) is 24
+        or len(selection) is 0
+    ):
+        return firstOutput, (datePicked, " - showing hour(s): All")
+
+    for x in selection:
+        holder.append(int(x))
+    holder.sort()
+
+    if holder[len(holder) - 1] - holder[0] + 2 == len(holder) + 1 and len(holder) > 2:
+        return (
+            firstOutput,
+            (
+                datePicked,
+                " - showing hour(s): ",
+                holder[0],
+                "-",
+                holder[len(holder) - 1],
+            ),
+        )
+
+    x = ""
+    for h in holder:
+        if holder.index(h) == (len(holder) - 1):
+            x += str(h)
+        else:
+            x += str(h) + ", "
```

Comment:
You can replace line `270-275`:

`holder_to_string = ", ".join(str(x) for x in holder)`